include/libp2p/common/metrics/instance_count.hpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_COMMON_METRICS_INSTANCE_COUNT_HPP
+#define LIBP2P_COMMON_METRICS_INSTANCE_COUNT_HPP
+
+#include <atomic>
+#include <mutex>
+#include <string_view>
+#include <unordered_map>
+
+#define _LIBP2P_METRICS_INSTANCE_COUNT(type, key)                    \
+  struct Libp2pMetricsInstanceCount {                                \
+    static auto &count() {                                           \
+      static auto &count = ([]() -> auto & {                         \
+        auto &state{libp2p::metrics::instance::State::get()};        \
```

Comment:
```suggestion
        auto &state{::libp2p::metrics::instance::State::get()};        \
```
name clashing is possible for "fc::libp2p::"

Reply:
That project does not know anything about fc namespace, thus it is perfectly fine to use just libp2p